core/types/block_ext_test.go

```diff
@@ -240,98 +329,55 @@ func TestBlockExtraRLP(t *testing.T) {
 
 	extra := &BlockBodyExtra{
 		Version: 10,
-		ExtData: ptrTo([]byte{11}),
+		ExtData: &[]byte{11},
 	}
 	extras.Block.Set(block, extra)
 
-	b, err := rlp.EncodeToBytes(block)
+	encoded, err := rlp.EncodeToBytes(block)
 	require.NoError(t, err)
 
 	gotBlock := new(Block)
-	err = rlp.DecodeBytes(b, gotBlock)
+	err = rlp.DecodeBytes(encoded, gotBlock)
 	require.NoError(t, err)
 
-	// Check transactions in the following because unexported fields
-	// size and time are set by the RLP decoder.
-	gotTxs := gotBlock.Transactions()
-	require.Len(t, gotTxs, 1)
-	gotTx := gotTxs[0]
-	gotTxBin, err := gotTx.MarshalBinary()
-	require.NoError(t, err)
-	wantTxBin, err := testTx.MarshalBinary()
-	require.NoError(t, err)
-	assert.Equal(t, wantTxBin, gotTxBin)
+	wantBlock := block.WithWithdrawals(nil)
 
-	wantHeader := &Header{
-		ParentHash: common.Hash{1},
-		Extra:      []byte{}, // field set by RLP decoder
-	}
-	wantHeaderExtras := &HeaderExtra{ExtDataHash: common.Hash{2}}
-	SetHeaderExtra(wantHeader, wantHeaderExtras)
-
-	wantUncle := &Header{
-		ParentHash: common.Hash{6},
-		Difficulty: new(big.Int), // field set by RLP decoder
-		Number:     new(big.Int), // field set by RLP decoder
-		Extra:      []byte{},     // field set by RLP decoder
+	opts := cmp.Options{
+		txHashComparer(),
+		headerHashComparer(),
+		cmpopts.IgnoreUnexported(Block{}),
 	}
-	wantUncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
-	SetHeaderExtra(wantUncle, wantUncleExtra)
-	wantUncles := []*Header{wantUncle}
-
-	wantBlock := NewBlock(wantHeader, gotTxs /* checked above */, wantUncles, receipts, stubHasher{})
-	wantExtra := &BlockBodyExtra{
-		Version: 10,
-		ExtData: ptrTo([]byte{11}),
+	if diff := cmp.Diff(wantBlock, gotBlock, opts); diff != "" {
+		t.Errorf("%T diff after RLP round-trip (-want +got):\n%s", gotBlock, diff)
 	}
-	extras.Block.Set(wantBlock, wantExtra)
-	_ = wantBlock.Size() // set block cached unexported size field
 
-	assert.Equal(t, wantBlock, gotBlock)
 	gotExtra := extras.Block.Get(gotBlock)
-	assert.Equal(t, wantExtra, gotExtra)
+	wantExtra := extra
+	if diff := cmp.Diff(wantExtra, gotExtra); diff != "" {
+		t.Errorf("%T diff after RLP round-trip of %T (-want +got):\n%s", wantExtra, wantBlock, diff)
+	}
 }
 
 // TestBlockBody tests the [BlockBodyExtra.Copy] method is implemented correctly.
 func TestBlockBody(t *testing.T) {
 	t.Parallel()
 
-	header := &Header{ParentHash: common.Hash{1}}
-	tx := NewTransaction(0, common.Address{1}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
-	txs := []*Transaction{tx}
-	uncles := []*Header{{ParentHash: common.Hash{6}}}
-	receipts := []*Receipt{{PostState: []byte{7}}}
-	block := NewBlock(header, txs, uncles, receipts, stubHasher{})
 	blockExtras := &BlockBodyExtra{
-		Version: 8,
-		ExtData: ptrTo([]byte{9}),
+		Version: 1,
+		ExtData: &[]byte{2},
```

Comment:
These were deliberately made to be constants so that the reader of the code doesn't need to compare each field. Please revert the change.

This is an obviously correct declaration with only local information:
```go
_ = HeaderExtra{
  Version: version,
  ExtData: []byte{extDataByte},
}
```

Whereas this forces the reader to compare fields across the declarations (i.e. non-local information), opening up to errors:
```go
_ = HeaderExtra{
  Version: 1,
  ExtData: []byte{2},
}
```

Reply:
I thought it would be easier to read a shorter code and that it was trivial to compare 😄 
I did revert it but with a version constant + `extData` variable set to `&[]byte{2}` - otherwise the reader would still have to check the expression is the same for both ExtData fields imo.